ci/infra/testrunner/skuba/skuba.py

```diff
@@ -45,27 +46,42 @@ def cleanup(conf):
         Utils.cleanup_files(dirs)
 
     @step
-    def cluster_init(self, kubernetes_version=None):
+    def cluster_init(self, kubernetes_version=None, cloud_provider=None):
         logger.debug("Cleaning up any previous test-cluster dir")
         self.utils.cleanup_file(self.cwd)
 
-        k8s_version_option = ""
+        k8s_version_option, cloud_provider_option = "", ""
         if kubernetes_version:
             k8s_version_option = "--kubernetes-version {}".format(kubernetes_version)
-        cmd = "cluster init --control-plane {} {} test-cluster".format(self.platform.get_lb_ipaddr(), k8s_version_option)
+        if cloud_provider:
+            cloud_provider_option = "--cloud-provider {}".format(cloud_provider)
+
+        cmd = "cluster init --control-plane {} {} {} test-cluster".format(self.platform.get_lb_ipaddr(),
+                                                                          k8s_version_option, cloud_provider_option)
         # Override work directory, because init must run in the parent directory of the
         # cluster directory
         self._run_skuba(cmd, cwd=self.conf.workspace)
 
     @step
-    def node_bootstrap(self):
+    def node_bootstrap(self, cloud_provider=None):
         self._verify_bootstrap_dependency()
 
+        if cloud_provider:
+            self.setup_config_cpi(cloud_provider)
+
         master0_ip = self.platform.get_nodes_ipaddrs("master")[0]
-        cmd = "node bootstrap --user {username} --sudo --target \
-                 {ip} my-master-0".format(ip=master0_ip, username=self.conf.nodeuser)
+        cmd = "node bootstrap --user {nodeuser} --sudo --target \
+                 {ip} caasp-master-{username}-0".format(ip=master0_ip, nodeuser=self.conf.nodeuser,
+                                                        username=self.conf.username)
         self._run_skuba(cmd)
 
+    @step
+    def setup_config_cpi(self, cloud_provider):
+        if cloud_provider == "openstack":
```

Comment:
This can further improved the decoupling of skuba class from this logic if you implement the `setup_cloud_provider`method in the `Platform class and by default it return an error and overrides it in the openstack class, as you did:`
```
class Platform:

    def setup_cloud_provider():
          raise Exception("Cloud provider not supported for this platform")
```

Unfortunately, here we miss the name of the platform (because how Platform class is presently implemented) but I prefer this approach.